src/justfile.rs

```diff
@@ -28,16 +28,32 @@ impl<'src> Justfile<'src> {
     self.recipes.len()
   }
 
-  pub(crate) fn suggest(&self, name: &str) -> Option<&'src str> {
-    let mut suggestions = self
+  pub(crate) fn suggest(&self, name: &str) -> Option<(&'src str, Option<&'src str>)> {
+    let mut available: Vec<(&str, usize, Option<&str>)> = self
       .recipes
       .keys()
-      .map(|suggestion| (edit_distance(suggestion, name), suggestion))
-      .collect::<Vec<_>>();
-    suggestions.sort();
-    if let Some(&(distance, suggestion)) = suggestions.first() {
+      .map(|&suggestion| (suggestion, edit_distance(suggestion, name), None))
+      .collect();
+    let mut aliases: Vec<(&str, usize, Option<&str>)> = self
+      .aliases
+      .iter()
+      .map(|(&alias_name, alias)| {
+        (
+          alias_name,
+          edit_distance(alias_name, name),
+          Some(alias.target.name.lexeme()),
+        )
+      })
+      .collect();
+    available.append(&mut aliases);
+    available.sort_by(|a, b| a.1.cmp(&b.1));
+    if let Some(&(suggestion, distance, alias_for)) = available.first() {
       if distance < 3 {
-        return Some(suggestion);
+        return if let Some(target) = alias_for {
```

Comment:
It would be simpler (here, at least) to just conditionally add the "an alias of Y" string to aliases, and then not have to change this function's return type (and cascade that change) but that runs into fighting the borrow checker for creating a new `String` from `format!` inside this method, and then returning that, which doesn't adhere to the lifetime of the struct.

Reply:
It's a little more cumbersome, but I often write types that capture things like this. Here's a possibility:

```
struct Suggestion<'src> {
  name:   &'src str,
  target: Option<&'src str>,
}

impl<'src> Display for Suggestion<'src> {
  fn fmt(&self, f: &mut Formatter) -> fmt::Result {
    write!(f, "Did you mean `{}`", self.name)?;
    if let Some(target) = self.target {
      write!(f, ", an alias for `{}`", target)?;
    }
    write!(f, "?")?;

    Ok(())
  }
}
```

And then suggest returns an `Option<Suggestion<'src>>`, and anything that gets a suggestion can just `eprintln("{}", suggestion)`.